intel_extension_for_transformers/llm/runtime/graph/models/gptneox/gptneox.cpp

```diff
@@ -135,41 +140,70 @@ static bool gptneox_model_eval_internal(model_context& lctx, const model_token*
 
         cur = ne_add(ctx0, ne_repeat(ctx0, model.layers[il].attn[1], cur), cur);
       }
-
-      struct ne_tensor* Qcur = ne_cont(ctx0, ne_view_3d(ctx0, cur, n_embd / n_head, n_head, N, cur->nb[1] / n_head,
-                                                        cur->nb[1], 0 * sizeof(float) * n_embd / n_head));
-      struct ne_tensor* Kcur = ne_cont(ctx0, ne_view_3d(ctx0, cur, n_embd / n_head, n_head, N, cur->nb[1] / n_head,
-                                                        cur->nb[1], 1 * sizeof(float) * n_embd / n_head));
-      struct ne_tensor* Vcur = ne_cont(ctx0, ne_view_3d(ctx0, cur, n_embd / n_head, n_head, N, cur->nb[1] / n_head,
-                                                        cur->nb[1], 2 * sizeof(float) * n_embd / n_head));
+      struct ne_tensor* Qcur = ne_cont(ctx0, ne_view_3d(ctx0, cur, head_dim, n_head, N * batch_size,
+                                                        cur->nb[1] / n_head, cur->nb[1], 0 * sizeof(float) * head_dim));
+      struct ne_tensor* Kcur = ne_cont(ctx0, ne_view_3d(ctx0, cur, head_dim, n_head, N * batch_size,
+                                                        cur->nb[1] / n_head, cur->nb[1], 1 * sizeof(float) * head_dim));
+      struct ne_tensor* Vcur = ne_cont(ctx0, ne_view_3d(ctx0, cur, head_dim, n_head, N * batch_size,
+                                                        cur->nb[1] / n_head, cur->nb[1], 2 * sizeof(float) * head_dim));
 
       // using mode = 2 for GPT-NeoX mode
-      Qcur = ne_rope_inplace(ctx0, Qcur, n_past, n_rot, 2, 0);
-      Kcur = ne_rope_inplace(ctx0, Kcur, n_past, n_rot, 2, 0);
+      Qcur = ne_rope_inplace(ctx0, ne_reshape_4d(ctx0, Qcur, head_dim, n_head, N, batch_size), n_past, n_rot, 2, 0);
+      Kcur = ne_rope_inplace(ctx0, ne_reshape_4d(ctx0, Kcur, head_dim, n_head, N, batch_size), n_past, n_rot, 2, 0);
 
       // store key and value to memory
       {
-        Vcur = ne_transpose(ctx0, ne_reshape_2d(ctx0, Vcur, n_embd, N));
-
-        struct ne_tensor* k =
-            ne_view_1d(ctx0, kv_self.k, N * n_embd, (ne_element_size(kv_self.k) * n_embd) * (il * n_ctx + n_past));
-        struct ne_tensor* v =
-            ne_view_2d(ctx0, kv_self.v, N, n_embd, (n_ctx)*ne_element_size(kv_self.v),
-                       (il * n_ctx) * ne_element_size(kv_self.v) * n_embd + n_past * ne_element_size(kv_self.v));
-
-        ne_build_forward_expand(&gf, ne_cpy(ctx0, Kcur, k));
-        ne_build_forward_expand(&gf, ne_cpy(ctx0, Vcur, v));
+        // Vcur = ne_transpose(ctx0, ne_reshape_2d(ctx0, Vcur, n_embd, N * batch_size));
```

Comment:
remove

Reply:
fixed in https://github.com/intel/intel-extension-for-transformers/pull/449/commits/540d412fb16d89de60bca98765f4a1e4d231a4e3